scripts/core/editor3/components/MultipleHighlights.jsx

```diff
@@ -68,6 +72,10 @@ export class MultipleHighlights extends React.Component {
     }
 
     getHighlightTypeFromStyleName(styleName) {
+        if (typeof styleName !== 'string') {
+            throw new Error('string expected');
```

Comment:
What do you suggest we do? When deciding whenever to throw or not, I'm thinking about what will potentially happen with the returned value. For example, if you are trying to delete something which doesn't exist, no worries, we can just pretend it didn't happen and continue with the program.

On the other hand, when you are requesting a value, which you will likely try to manipulate or pass to other functions we should make sure that invalid values are stopped as soon as possible.

According to the current implementation, I don't think there are possibilities for that error to occur. Even if I returned `null`, somewhere down the line some code might try to do something with the value and would get an error anyway, so I thought it might be better to fail fast instead of 10 levels deeper into the call stack.

Reply:
I don't know. what about `console.assert` ? if this shouldn't happen it might be ok, but if we throw exception it might block the ui either in react or angular afaik

Reply:
Even if you use `console.assert` you will still have to return something from the function. And if you return `null`, that value will get manipulated or passed to other functions which will try to manipulate it which will cause an error to be thrown and block the UI.

I've checked the implementation, there is only one place calling the function and it validates the argument before calling. Anyway, I think it's better to break the UI right away than end up with a corrupted state which will also break the UI, only a few seconds later and it will be harder to trace.

Reply:
it's ok to throw exception asap, but we should handle that case somehow so we don't leave users with unusable ui. if it never happens then there is no point to add that check

Reply:
For now, it doesn't happen, but the code will change in the future and the function might end up being called with bad arguments. When it happens, I'd rather throw an error and break the UI, than return `null`, which might corrupt the state somewhere else and in turn will get an error thrown anyway.

I can return `null` if you think it's better, but I don't see the advantages.

Reply:
ok np. maybe we should integrate something like flow to handle the static type checking

Reply:
Yeah, static type checking would be awesome. The code would be cleaner from all the manual checks and it would give us better confidence in the correctness of it. TypeScript is also an option, which I personally prefer over flow since it can provide more guarantees.

Reply:
I think with flow we should get the type checking also for draft/react, but not sure about it. would you give it a shot to integrate it in this PR instead of doing the manual checks?

Reply:
With TypeScript, you would also get type checking for react and probably draft, but it may be more work to integrate it.

I'm currently working with Pablo on implementing the history for comments/annotations and don't want to block his work, so I want to finish that first and I could then get back and fix this one.

I guess we can leave the PR unmerged and will get back to it later if it's ok with you.